docs/repair_based_node_ops.md

```diff
@@ -0,0 +1,60 @@
+# Repair based node operations
+
+Here below is a simple introduction to the node operations Scylla supports and
+some of the issues.
+
+- Replace operation
+
+   It is used to replace a dead node. The token ring does not change. Replacing
+   node pulls data from only one of the replicas which might not be the latest
+   copy.
+
+- Rebuild operation
+
+   It is used to get all the data this node owns form other existing nodes. It
+   pulls data from only one of the replicas which might not be the latest copy.
+
+- Bootstrap operation
+
+   It is used to add a new node into the cluster. The token ring changes. Do no
+   suffer from the "not the latest replica” issue. New node pulls data from
+   existing nodes that are losing the token ranges.
+
+   It suffers from failed streaming. We split the ranges in 10 groups and we
+   stream one group at a time. Restream the group if failed, causing
+   unnecessary data transmission on wire.
+
+   Bootstrap is not resumable. With failure after 99.99% of data is streamed,
+   if we restart the node again, we need to stream all the data again even if
+   the node already has 99.99% of the data.
+
+- Decommission operation
+
+   It is used to remove a live node form the cluster. Token ring changes. It
+   does not suffer “not the latest replica” issue. The leaving node pushes data
+   to existing nodes.
+
+   It suffers from resumable issue like bootstrap operation.
+
+- Removenode operation
+
+   It is used to remove a dead node out of the cluster. Existing nodes pull
+   data from other existing nodes for the new ranges it owns. It pulls from one
+   of the replicas which might not be the latest copy.
+
+To solve all the issues above. We could use repair based node operation.  The
+idea behind repair based node operations is simple: use repair to sync data
+between replicas instead of streaming.
+
+The benefits:
+
+- Latest copy is guaranteed
+
+- Resumable in nature
+
+- No extra data is streamed on wire
+  E.g., rebuild twice, will not stream the same data twice
+
+- Unified code path for all the node operations
+
+- Free repair operation during bootstrap, replace operation and so on.
```

Comment:
Please add an explanation about the node selection algorithms that I asked about in the previous iteration. It's easier to review in a document, and will be needed for later reference.

Reply:
OK. I will add one.

Reply:
Can you please take a look at this comments: https://github.com/scylladb/scylla/pull/4975#discussion_r362445651